build.sbt

```diff
@@ -107,6 +110,7 @@ lazy val sqs = project
   .settings(
     libraryDependencies ++= Seq(
       Dependencies.Libraries.sqs,
+      Dependencies.Libraries.sts,
```

Comment:
add STS to fix up AWS default auth chain for K8s

Reply:
Please can you help me understand this one better?  I am familiar with STS and Kubernetes (although not together!).  But I cannot find any documentation that describes the default auth chain specifically on Kubernetes.

I used [this page](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html) to understand the default auth chain in other contexts, but it does not mention STS.  Do you see an error message if this lib is not included?  What is the error message?  How are you providing credentials to your collector?

I'm not against including this in the next release, if I can understand it better.

Reply:
I started to type out a detailed explanation but then I decided it was probably better if I let Amazon explain it which they do [here](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/) and [here](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html).

Essentially that third option "Web Tokens" allows us to create [Kubernetes Service Accounts](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/) that are tied to IAM roles. When those roles are assigned to pods then anything running the AWS API in those pods will use assume that IAM role. This is easily done as part of a standard deployment. The details are spelled out in links above but long story short it uses OIDC, Web Tokens, a service on your K8s cluster, and STS to make it work. 

In the collector container without the STS library (and you have enough levels of debugging enabled) you can actually see a warning in the console that tells you that the "Web Tokens" method was skipped because the STS library is not on the class path. The problem then becomes that container then falls back to the IAM role that the node (EC2 Instance role) has. This is less than ideal for a variety of reasons, not least the lease of which is you don't always want every container running on a node to have access to the same things. This [StackExchange post](https://stackoverflow.com/questions/67175006/snowplow-collector-on-k8s-does-not-use-service-account) spells out the issue pretty well. I was having the same issue which is how I found that post and what set me down the path that ultimately led to this PR.


Hopefully that answers the question, but let me know if you need more :) 

P.S.
If you or anyone are looking for helm charts / manifests for deploying a snowplow pipeline in Kubernetes hit me up, I have a set that works on AWS to deploy the collector, enrich, s3 loader (raw, good, bad), iglu server, postgres loader (enriched, bad), and dataflow runner and am happy to share if they are of interest.

Reply:
That is a very helpful explanation, and helpful links - thank you very much indeed.  I agree we should get this change merged, and we should do something similar in other snowplow apps going forwards.

I think here you have added the lib only to the "sqs" collector, but actually we should add it to the "kinesis" collector also.

Reply:
> If you or anyone are looking for helm charts / manifests for deploying a snowplow pipeline in Kubernetes hit me up


Yes Please! :D
My email is
c
a
l
e
b
@
1
5
f
i
v
e
.
c
o
m

(email obfuscated to avoid spambots)